spec/system/post_index_spec.rb

```diff
@@ -0,0 +1,26 @@
+# post_index_spec.rb
+require 'rails_helper'
+
+RSpec.feature 'User Post Index', type: :feature do
+  before do
+    # Set up test data as needed
+    @user = FactoryBot.create(:user)
+    @posts = FactoryBot.create_list(:post, 10, author: @user)
+    visit user_posts_path(@user)
+  end
+
+  scenario 'displays post index page content' do
+    expect(page).to have_css('.user-container')
+    expect(page).to have_css('.img-container img')
+    expect(page).to have_css('.user-name', text: @user.name)
+
+    @posts.each do |post|
+      expect(page).to have_css('.post', text: post.title)
+      expect(page).to have_css('.post-content p', text: post.text)
+      expect(page).to have_css('.interactions', text: "comments: #{post.comments_counter}")
+      expect(page).to have_css('.interactions', text: "likes: #{post.likes_counter}")
+    end
+
+    expect(page).to have_css('.pagination') # Check for pagination section
```

Comment:
- [ ] Hello, You have done a good job so far. I would suggest you add these tests you left behind: 
- Tests that asserts you can see the number of posts the user has written.
- Tests that asserts you can see the first comments on a post.
- Tests that redirects to the post's show page when you click on a post.
I am requesting this because you need to test all stories provided (Project requirements)